### PR TITLE
fix broken tests and fix v0.6 deprecations

### PR DESCRIPTION
**Pull request by rdeits**

### CODE REVIEW

File: src/ObjectiveFunctions.jl [hunk 1]
```diff
@@ -41,20 +42,20 @@ immutable LossTransform{T,L<:Loss} <: AbstractLossTransform{T}
...
+        new{T, L}(loss, nin, input, target, output)
     end
 end
 
 LossTransform{L<:Loss}(loss::L, nin::Int) = LossTransform{Float64, L}(loss, nin)
 
 # input and target are pre-populated... compute the output value as: ∑ loss(targetᵢ, inputᵢ)
 function transform!(lt::LossTransform)
-    lt.output.val[1] = sumvalue(lt.loss, value(lt.target), input_value(lt))
+    lt.output.val[1] = value(lt.loss, value(lt.target), input_value(lt), AvgMode.Sum())
```
Comment (rdeits) [inline src/ObjectiveFunctions.jl:58]: I changed this based on https://github.com/JuliaML/LossFunctions.jl/commit/774014d95fff62fbe439d8b08ae023e5cdb57aec#diff-f2fc3674e12bf71a707cb4988076d39eR9

**Comment by rdeits** [comment]
I also updated Travis to add v0.6 and remove osx testing (OSX testing is slow and, IMO, not super useful for pure-julia packages like this).

**Review by Evizero — APPROVED** [review]
lgtm. We can reactive OSX when the package is more stable.

**Comment by Evizero** [comment]
I'll merge once CI passes. Thanks!

File: REQUIRE [hunk 2]
```diff
@@ -2,3 +2,7 @@ julia 0.5-
 
 Reexport
 LearnBase
+LossFunctions
+PenaltyFunctions
+Transformations
```
Comment (Evizero) [inline REQUIRE]: Is `Transformations` in METADATA? I doubt it. We may need to put a `Pkg.clone` in `.travis.yml` here https://github.com/JuliaML/ObjectiveFunctions.jl/blob/master/.travis.yml#L14 instead

Comment (Evizero) [reply REQUIRE]: you beat me to it. nice!

Comment (rdeits) [reply REQUIRE]: Yup, just realized that.

**Comment by rdeits** [comment]
I am trapped in a nightmare world of cyclic dependencies between this package, StochasticOptimization, and Transformations :frowning:

**Comment by rdeits** [comment]
Also, PenaltyFunctions is 0.6-only, so I'm bumping up the REQUIRE and travis versions

**Comment by rdeits** [comment]
Next blocker: https://github.com/JuliaML/Transformations.jl/pull/5

**Comment by Evizero** [comment]
Sorry that this is frustrating. Know that your effort is appreciated

**Comment by rdeits** [comment]
Hey, it's still better than *my* research code :wink:

**Comment by rdeits** [comment]
I also have high hopes that Pkg3 will make managing this kind of distributed ecosystem much easier, since a lot of this trouble could be avoided if it were easier for us to clearly specify which versions of each of these packages worked well together. As it is, I've had to use some hack-y tools to manage packages pointing to specific git SHAs in my own work.

**Comment by rdeits** [comment]
Woohoo! One down!

**Comment by Evizero** [comment]
Thanks!

**Pull request merged**